solr/contrib/ltr/src/java/org/apache/solr/ltr/model/LTRScoringModel.java

```diff
@@ -16,73 +16,217 @@
+      if (feature == null) {
+        if (other.feature != null) return false;
+      } else if (!feature.equals(other.feature)) return false;
+      if (normalizer == null) {
+        if (other.normalizer != null) return false;
+      } else if (!normalizer.equals(other.normalizer)) return false;
+      return true;
+    }
+
+    @Override
+    public FeatureWeight createFeatureWeight(IndexSearcher searcher, boolean needsScores, SolrQueryRequest request,
+        Query originalQuery, Map<String,String[]> efi) throws IOException {
+      return feature.createFeatureWeight(searcher, needsScores, request, originalQuery, efi);
+    }
+
+    @Override
+    public int getIndex() {
+      return feature.getIndex();
+    }
+
+  }
+
   protected final String name;
   private final String featureStoreName;
-  protected final List<Feature> features;
+  protected final List<ModelFeature> modelFeatures;
   private final List<Feature> allFeatures;
   private final Map<String,Object> params;
-  private final List<Normalizer> norms;
 
+  public static LTRScoringModel getInstance(SolrResourceLoader solrResourceLoader,
+      String className, String name, List<ModelFeature> modelFeatures,
+      String featureStoreName, List<Feature> allFeatures,
+      Map<String,Object> params) throws ModelException {
+    final LTRScoringModel model;
+    try {
+      // create an instance of the model
+      model = solrResourceLoader.newInstance(
+          className,
+          LTRScoringModel.class,
+          new String[0], // no sub packages
+          new Class[] { String.class, List.class, String.class, List.class, Map.class },
+          new Object[] { name, modelFeatures, featureStoreName, allFeatures, params });
+      if (params != null) {
+        SolrPluginUtils.invokeSetters(model, params.entrySet());
+      }
+    } catch (final Exception e) {
+      System.out.println("cpoerschke debug: caught "+e);
+      throw new ModelException("Model type does not exist " + className, e);
+    }
+    model.validate();
+    return model;
+  }
+
+  @Deprecated
+  public static LTRScoringModel getInstance(SolrResourceLoader solrResourceLoader,
+      String className, String name, List<Feature> features,
+      List<Normalizer> norms,
+      String featureStoreName, List<Feature> allFeatures,
+      Map<String,Object> params) throws ModelException {
+    final LTRScoringModel model;
+    try {
+      // create an instance of the model
+      model = solrResourceLoader.newInstance(
+          className,
+          LTRScoringModel.class,
+          new String[0], // no sub packages
+          new Class[] { String.class, List.class, List.class, String.class, List.class, Map.class },
+          new Object[] { name, features, norms, featureStoreName, allFeatures, params });
+      if (params != null) {
+        SolrPluginUtils.invokeSetters(model, params.entrySet());
+      }
+    } catch (final Exception e) {
+      System.out.println("cpoerschke debug: caught "+e);
```

Comment:
prints needed? if so switch to logger instead (for other locations too if they exist)